src/RokuAdapter.ts

```diff
@@ -1128,43 +1165,52 @@ export class RequestPipeline {
                     this.emit('console-output', command);
                 }
                 this.client.write(commandText);
+                if (waitForPrompt) {
+                    this.isAtDebuggerPrompt = false;
+                }
             };
-            this.requests.push({
+
+            let request = {
                 executeCommand: executeCommand,
                 onComplete: (data) => {
                     console.debug(`Command finished (${waitForPrompt ? 'after waiting for prompt' : 'did not wait for prompt'}`, command);
                     console.debug('Data:', data);
                     resolve(data);
                 },
-                waitForPrompt: waitForPrompt
-            });
-            //start processing (safe to call multiple times)
-            this.process();
+                waitForPrompt: waitForPrompt,
+            };
+
+            if (!waitForPrompt) {
+                if (!this.isProcessing) {
+                    //fire and forget the command
+                    request.executeCommand();
+                    //the command doesn't care about the output, resolve it immediately
+                    request.onComplete(undefined);
+                }
+            } else {
+                this.requests.push(request);
+                if (this.isAtDebuggerPrompt) {
```

Comment:
Same issue as above. What happens when `waitForPrompt === true && this.isAtDebuggerPrompt === false`